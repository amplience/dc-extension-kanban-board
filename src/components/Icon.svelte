<script lang="ts">
  export let icon: string;
  export let className = '';
  export let width = '24px';
  export let height = '24px';
</script>

<style>
  div {
    display: inline-block;
    width: var(--icon-width);
    height: var(--icon-height);
    color: #333;
  }

  div :global(svg) {
    width: var(--icon-width);
    height: var(--icon-height);
  }
</style>

<div class={className} style="--icon-width: {width}; --icon-height: {height}">
  {@html icon}
</div>
