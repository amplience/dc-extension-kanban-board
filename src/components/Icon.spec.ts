import { render } from '@testing-library/svelte';
import Icon from './Icon.svelte';

describe('Icon', () => {
  it('should render a Icon', () => {
    const iconMock = '<svg><text>svg text</text></svg>';
    const { container } = render(Icon, { icon: iconMock });
    expect(container.firstChild).toMatchSnapshot();
  });
});
