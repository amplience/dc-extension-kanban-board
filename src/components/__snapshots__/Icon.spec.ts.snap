// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render a Icon 1`] = `
<div>
  <div
    class=" svelte-zpbvvv"
    style="--icon-width: 24px; --icon-height: 24px;"
  >
    <svg>
      <text>
        svg text
      </text>
    </svg>
  </div>
</div>
`;
